docroot/modules/humsci/hs_blocks/src/Plugin/Block/SocialMediaBlock.php

```diff
@@ -0,0 +1,386 @@
+        ['@user_guide_url' => 'https://hsweb.slite.page/p/NeJL89GqNsiOY-/Social-Media-Footer-block']
+      ),
+      '#add_more_label' => $this->t('Add another item'),
+      '#element_validate' => [
+        [get_class($this), 'validateLinks'],
+      ],
+      '#attributes' => [
+        'id' => 'links-wrapper',
+      ],
+    ];
+
+    foreach ($form_state->get('links') as $key => $link) {
+      $form['links'][$key] = [
+        '#type' => 'container',
+        'link_url' => [
+          '#type' => 'url',
+          '#title' => $this->t('URL'),
+          '#description' => $this->t('Social Media Profile URL.'),
+          '#default_value' => $link['link_url'],
+        ],
+        'link_title' => [
+          '#type' => 'textfield',
+          '#title' => $this->t('Label'),
+          '#description' => $this->t('If empty, the social platform name will be used for supported platforms, otherwise the domain name will be used.'),
+          '#default_value' => $link['link_title'],
+        ],
+        '_weight' => [
+          '#type' => 'weight',
+          '#title' => $this->t('Weight'),
+          '#title_display' => 'invisible',
+          '#delta' => count($this->configuration['links']),
+          '#default_value' => $link['_weight'],
+        ],
+      ];
+    }
+
+    $form['links']['add_more'] = [
+      '#type' => 'submit',
+      '#name' => 'links_add_more',
+      '#value' => $form['links']['#add_more_label'],
+      '#submit' => [[get_class($this), 'addMoreSubmit']],
+      '#ajax' => [
+        'callback' => [get_class($this), 'addMoreAjax'],
+        'wrapper' => 'links-wrapper',
+        'effect' => 'fade',
+      ],
+    ];
+
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function blockSubmit($form, FormStateInterface $form_state): void {
+    $this->configuration['icon_size'] = $form_state->getValue('icon_size');
+    $this->configuration['layout'] = $form_state->getValue('layout');
+
+    // Only save links if they're not empty.
+    $links = array_filter($form_state->getValue('links'), function ($link) {
+      return !empty($link['link_url']);
+    });
+    $this->configuration['links'] = $links;
+
+    // This sets the placed block ID to be used for a custom contextual link.
+    $this->configuration['placed_block_id'] = $form['id']['#default_value'];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function build(): array {
+    $links = array_map([$this, 'linkWithIcon'], $this->configuration['links']);
+    $placed_block_id = $this->configuration['placed_block_id'];
+
+    $build = [
+      '#theme' => 'hs_blocks_social_media',
+      '#icon_size' => $this->configuration['icon_size'],
+      '#layout' => $this->configuration['layout'],
+      '#links' => $links,
+      '#cache' => [
+        'tags' => array_merge($this->getCacheTags(), ['block_view']),
+        'contexts' => ['user', 'user.permissions'],
+      ],
```

Comment:
The block itself contains the appropriate tags and contexts. this shouldn't be necessary.

Reply:
Fixed.